scrapers_next/il/people.py

```diff
@@ -0,0 +1,206 @@
+        )
+        additional_office, additional_phone = additional_office.split("(")
+        # print(additional_office)
+        # print(additional_phone)
+
+        p.additional_offices = additional_office
+        p.extras["extra phone"] = additional_phone
+
+
+class LegDetail(HtmlPage):
+    def process_page(self):
+        p = self.input
+
+        img = CSS("body table tr td img").match(self.root)[2].get("src")
+        p.image = img
+
+        (
+            capitol_addr,
+            capitol_phone,
+            capitol_fax,
+            district_addr,
+            district_phone,
+            district_fax,
+            email,
+            # additional_office,
+            # additional_phone
+        ) = self.process_addrs(p)
+
+        p.capitol_office.address = capitol_addr
+        p.capitol_office.voice = capitol_phone
+        if capitol_fax:
+            p.capitol_office.fax = capitol_fax
+        if district_addr:
+            p.district_office.address = district_addr
+        if district_phone:
+            p.district_office.voice = district_phone
+        if district_fax:
+            p.district_office.fax = district_fax
+        if email:
+            p.email = email
+        # if additional_office:
+        #     p.additional_address = additional_office
+        # if additional_phone:
+        #     p.extras["extra phone"] = additional_phone
+
+        return p
+
+    def process_addrs(self, p):
+        """
+        For a given detail page, len(CSS("body table tr td.member").match(self.root))
+        can be either 7, 10, 11, 12, 13, 14, or 15. This function / for loop handles
+        pages with varies number of 'address' lines.
+        """
+        (
+            capitol_addr,
+            cap_phone,
+            capitol_fax,
+            district_addr,
+            dis_phone,
+            dis_fax,
+            email,
+            # additional_office,
+            # additional_phone
+        ) = (None, None, None, None, None, None, None)
+        addresses = CSS("body table tr td.member").match(self.root)
+        line_number = 0
+        district_addr_lines = 0
+        for line in addresses:
+            if line.text_content().strip() == "Additional District Addresses":
+                print("ADDITIONAL ADDRS")
+                # additional_addr_link = line.getchildren()[0].get("href")
+                # print(additional_addr_link)
+                # AdditionalAddresses(p, source=additional_addr_link)
+
+            line = line.text_content().strip()
+            if (
+                line == "Springfield Office:"
+                or line == "District Office:"
+                or line == ""
+                or line == "Additional District Addresses"
+                or line.startswith("Senator")
+                or line.startswith("Representative")
+                or line.startswith("Years served:")
+            ):
+                line_number += 1
```

Comment:
does line_number get incremented no matter what?  if so maybe we do it at the top instead of each if conditional needing one 

also check out `enumerate`: https://realpython.com/python-enumerate/ potentially